### ISSUE TITLE
Extract NexusClient as internal field in AbstractNexusStagingRepositoryTask

### ISSUE THREAD

**Issue by szpak** [gradle-nexus/publish-plugin#69]
I believe, I proposed it in one of my earlier comments, but suggested to move it to a separate PR, after this one is merged to do not mix the things.

_Originally posted by @szpak in https://github.com/gradle-nexus/publish-plugin/pull/62#r579827405_

After #62 is merged.

**Comment by TWiStErRob** [comment]
A step towards, or fixed by https://github.com/gradle-nexus/publish-plugin/pull/252/commits/0c1bf8bbacea50e1bf2351ec8b8f0c6a82f5bb03

**Comment by szpak** [comment]
I think it could be closed now.
